core/src/main/java/org/vertexium/query/Query.java

```diff
@@ -10,20 +10,76 @@
 
     QueryResultsIterable<Vertex> vertices(EnumSet<FetchHint> fetchHints);
 
+    /**
+     * Execute the query and return the ids of all matching vertices.
+     * This method will not load the data from storage, which is more
+     * efficient in cases where you only need to know the ids of matching vertices.
+     * Hidden vertices are not included in the results.
+     *
+     * @return The ids of vertices that match this query.
+     */
     QueryResultsIterable<String> vertexIds();
 
+    /**
+     * Execute the query and return the ids of all matching vertices.
+     * This method will not load the data from storage, which is more
+     * efficient in cases where you only need to know the ids of matching vertices.
+     *
+     * @param includeHidden Whether or not to include hidden vertices in the results
+     * @return The ids of vertices that match this query.
+     */
+    QueryResultsIterable<String> vertexIds(boolean includeHidden);
```

Comment:
I don't love the mystery boolean parameter. It feels a little weird to pass `FetchHint`s here though. Maybe `IdFetchHint`s which could be an `EnumSet` or just an `enum` with two values? Not really sure.